### PR TITLE
[ADD]account_analytic_parent

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
Analytic Analytic Parent
========================

This module reintroduces the hierarchy to the analytic accounts as it was in previous versions of Odoo. This module is a base module for other modules to manage the hierarchy concept in analytics.

There is an opened issue regarding this: https://github.com/OCA/account-analytic/issues/65

### CODE REVIEW

**Review by elicoidal — APPROVED** [review]
nice!

**Comment by elicoidal** [comment]
no functional tests

File: account_analytic_parent/models/account_analytic_account.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        of analytic account changes
+
+        """
+        for account in self:
+            super(AccountAnalyticAccount, self)._compute_debit_credit_balance()
+            debit = account.debit
+            credit = account.credit
+            balance = account.balance
+            children = account.search([('parent_id', 'child_of', account.id),
+                                       ('id', '!=', account.id)])
```
Comment (LoisRForgeFlow) [inline account_analytic_parent/models/account_analytic_account.py]: Why don't you use the <code>child_ids</code> you've declared at the beginning??

File: account_analytic_parent/models/account_analytic_account.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    @api.onchange('parent_id')
+    def on_change_parent(self):
+        for account in self:
+            if account.parent_id:
+                parent = account.parent_id
+                partner = parent.partner_id and parent.partner_id.id or False
+            else:
+                partner = False
+            if partner:
+                account.partner_id = partner
```
Comment (LoisRForgeFlow) [inline account_analytic_parent/models/account_analytic_account.py]: <code>account.partner_id = account.parent_id.partner_id or False</code> after the for loop is enough, isn't it?

Comment (AaronHForgeFlow) [reply account_analytic_parent/models/account_analytic_account.py]: yes it is

File: account_analytic_parent/models/account_analytic_account.py (hunk 1 above)
Comment (AaronHForgeFlow) [reply account_analytic_parent/models/account_analytic_account.py]: I agree. Thanks :1st_place_medal:

**Comment by AaronHForgeFlow** [comment]
Added some tests. Thanks @elicoidal :ok_hand:

**Review by LoisRForgeFlow — APPROVED** [review]
This is perfect now  :white_check_mark:

**Review by JordiBForgeFlow — APPROVED** [review]
:+1: Technical and functional review

**Comment by JordiBForgeFlow** [comment]
@elicoidal Are you good with the changes? This one is ready to merge.

**Pull request merged**